packages/http/src/utils/operations.ts

```diff
@@ -0,0 +1,55 @@
+import { transformOas3Operations } from '@stoplight/http-spec/oas3/operation';
+import { transformOas2Operations } from '@stoplight/http-spec/oas2/operation';
+import { transformPostmanCollectionOperations } from '@stoplight/http-spec/postman/operation';
+import * as $RefParser from '@stoplight/json-schema-ref-parser';
+import { HTTPResolverOptions } from '@stoplight/json-schema-ref-parser';
+import { bundleTarget, decycle } from '@stoplight/json';
+import { IHttpOperation } from '@stoplight/types';
+import { get } from 'lodash';
+import * as os from 'os';
+import type { Spec } from 'swagger-schema-official';
+import type { OpenAPIObject } from 'openapi3-ts';
+import type { CollectionDefinition } from 'postman-collection';
+
+export async function getHttpOperationsFromSpec(specFilePathOrObject: string | object): Promise<IHttpOperation[]> {
+  const prismVersion = require('../../package.json').version;
+  const httpResolverOpts: HTTPResolverOptions = {
+    headers: {
+      'User-Agent': `PrismMockServer/${prismVersion} (${os.type()} ${os.arch()} ${os.release()})`,
+    },
+  };
+  const result = decycle(
+    await new $RefParser().dereference(specFilePathOrObject, { resolve: { http: httpResolverOpts } })
```

Comment:
@daniel-white we are going to leave this here and then add a check for unresolved external references in our usage of this code and error at that point.